src/host/multiple_ledger.h

```diff
@@ -0,0 +1,805 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "consensus/ledger_enclave_types.h"
+#include "ds/logger.h"
+#include "ds/messaging.h"
+
+#include <cstdint>
+#include <cstdio>
+#include <filesystem>
+#include <limits>
+#include <list>
+#include <map>
+#include <string>
+#include <sys/types.h>
+#include <unistd.h>
+#include <vector>
+
+namespace fs = std::filesystem;
+
+namespace asynchost
+{
+  static constexpr size_t max_chunk_threshold_size =
+    std::numeric_limits<uint32_t>::max(); // 4GB
+  static constexpr size_t max_read_cache_size_default = 5;
```

Comment:
I presume this is entries, as opposed to bytes? If so, I suggest max_read_cache_entries.

Reply:
This is actually the number of files open. I'll clarify.

Reply:
That's what I meant by files, but you're right that files is better.